timeout_config.hh

```diff
@@ -10,20 +10,60 @@
 #pragma once
 
 #include "db/timeout_clock.hh"
+#include "utils/updateable_value.hh"
+#include <chrono>
+
+namespace db { class config; }
+
+class updated_timeout_config;
 
 struct timeout_config {
```

Comment:
This struct is left because things like client_state (and thrift_handler, but nobody cares about thrift nowadays) carry _it_ onboard, not the updated_timeout_config. So the whole timeout-config-thing becomes life-updateable in a sense that _new_ connections obey that, while old/existing ones continue working with older timeouts.

I think it deserves at least a comment somewhere in timeout_config.hh, because it's not immediately obvious that this is how things should work.

Reply:
good point. i made some notes in the comment. but this should be reflected in the code as well.

Reply:
Thanks!